### ISSUE TITLE
Needs updates for Ubuntu 20.04

### ISSUE THREAD

**Issue by StuartIanNaylor** [linto-ai/linto-desktoptools-hmg#2]
Its no problem as going to downgrade back to 18.04 as its 20.04 is still very fresh.

The Hotword Model Generator is excellent though and was quite shocked at the speed it managed to create reasonable working models. (Didn't even have the Cuda install so it was even CPU).

With hmg the problem is not now creating models but getting datasets.
Is there any chance you could create a tool to strip words out of the CommonVoice dataset.
I have used Deepspeech with the `--json` which gives start time and duration in a json file of all words recoginsed.
Then guess sox could split ASR sentences and pad into centered 1sec KWS words.

The metadata in the CommonVoice dataset is pretty awfull and its such a shame that they didn't enforce Gender, Age and rather than Accent asked for Dialect.
Such as https://openslr.org/83/ (Data set which contains male and female recordings of English from various dialects of the UK and Ireland.)
Is there any chance you guys could create another tool that can automate much of word stripping as you can drop results that don't verify to the validated text as the dataset is huge and will still give a huge wordset.
In fact there isn't just common voice there are quite a few 'ASR' datasets that could be word harvested that could give a massive resultantant KWS dataset.
The metadata is really crucial and you can hugely influence accuracy by tailouring to voice(s) of use and if your AI is mainly gender or dialect specific then your dataset should try to contain a majority of that as its dataset.

Also with Hotwords sylables are very apparent in MFCC images and in my own tensorflow models I was playing with additional classifications that where not hotwords but used as a weight to hot word detection.
From the Google command set I used the word 'Visual' as its 3 sylable then I organised 2 additional classifications of '1Sylable' & '2Sylable' with corresponding words.
My dataset as the google command set isn't huge but just from that simple test the score of additional classifications could be used to create a dynamic KW threshold where lower scores are accepted if equally low on the additional classification.
Actually looking with the hotword creation and folder selection that looks like it might already be implemented just haven't worked out the best way to progress.

Sylable gives a very clear MFFC imprint but also a relatively simple split as KW recommendations of 3 or more sylables means the majority of common words lie elsewhere.
Thought I would meantion that as it would be great if HMG was more flexible and maybe it already is but just wondred if that can be accomplished and also wondering if you find similar results and maybe use to improve accuracy?

Also is there anyway to automate preset noise and mix into the overall? Presuming the guasian noise does this or does that just create slight variation on the wav input?

PS HMG is really excellent

**Comment by StuartIanNaylor** [comment]
https://drive.google.com/file/d/1CYyacnAxnllwp9jfRkUZGTnoj9S6lvJw/view?usp=sharing

https://drive.google.com/file/d/1BJNRuTw0lRoM5Zy16Eowk2Q0s85sH4-C/view?usp=sharing

I have been using the Google command set word "visualise" due to phone count with HMG and from the above you can see after much tidying of the Google command set really good results can be attained.

0.117874333 false activation’s per hour isn't bad.

I am still using the binary version as struggling with the install and seem stick as after mfcc extraction I get an error that haven't been able to solve.

https://github.com/keras-team/autokeras/issues/1095

The evaluate to be able to test the data set is brilliantly presented as to be able to click and listen to  false negatives/positives for 
hotword/non-hotword gives instant feedback to your dataset quality anf potential errors.
It was this function that made me aware of how much in the Goggle command-set is actually bad and how much singular bad data can effect the overall data-set.
You can select all or individual wavs but there is no method to remove from data-set as I had to resort to filename searching and removing from disk.
Please add and 'edit' menu that when selected you can remove selected wav's from the dataset.

After doing this several times and removing maybe 7-8% of the google command set for 'visualise' as many where just completely wrongly trimmed with no word or words cut.
After removing so many bad my accuracy increased drastically and also my understanding on how bad input can effect the model.
Please add an 'Edit' menu with a 'Remove' when items are selected.

There are still a few crashed that I presume are unimplemented function as 'export test set' on that tab does, but is a minor detail.

The feature tab should of been brilliant to try what different parameters can do and if fine tuning to a specific language can help provide accuracy and also overall parameter increase for better phone detection at the expense of load or even reducing parameter count if possible.
Being able to analyse the samples and view result gives great feed back.
When you come to test the dataset any changes would seem to kill all accuracy even if only slight and really shouldn't.
I began to wondering when features are extracted for test if they don't use the MFCC manifest that created the model?

Just those 2 items would make HMG in that 80/20 rule fit for purpose and currently you can get it to work but its a bit of a struggle.
